internal/repository/repo.go

```diff
@@ -0,0 +1,117 @@
+package repository
+
+import (
+	model "github.com/cyneruxyz/address-book/gen/proto"
+	"hash/fnv"
+	"strings"
+	"sync"
+)
+
+type AddressBook struct {
+	sync.Mutex
```

Comment:
Mutex is not the most efficient solution here. Let's assume, we have a lot of readers, and very rare, episodic write operations. What could help our service to work better?